ckanext/unaids/custom_user_profile/actions.py

```diff
@@ -0,0 +1,157 @@
+
+    %(link)s
+
+    Best wishes,
+    The AIDS Data Repository
+    """)
+
+
+def mail_new_membership_request(locale, admin, group_name, url, user_name, user_email, user_job_title, user_affiliation):
+    """
+    Overrides the mail_new_membership_request function from ckanext-ytp-request plugin
+    """
+    with force_locale('en'):
+        subject = _SUBJECT_MEMBERSHIP_REQUEST % {
+                      'organization': group_name
+                  }
+        message = _MESSAGE_MEMBERSHIP_REQUEST % {
+                      'user': user_name,
+                      'email': user_email,
+                      'organization': group_name,
+                      'link': url,
+                      'job_title': user_job_title,
+                      'affiliation': user_affiliation
+                  }
+    try:
+        mail_user(admin, subject, message)
+    except Exception:
+        log.exception("Mail could not be sent")
+
+
+def _create_member_request(context, data_dict):
+    """
+    Whole method copied over from ckanext-ytp-request plugin
+    Helper to create member request
+    """
+    role = data_dict.get('role', None)
+    if not role:
+        raise logic.NotFound
+    group = model.Group.get(data_dict.get('group', None))
+
+    if not group or group.type != 'organization':
+        raise logic.NotFound
+
+    user = context.get('user', None)
+
+    if authz.is_sysadmin(user):
+        raise logic.ValidationError({}, {_("Role"): _(
+            "As a sysadmin, you already have access to all organizations")})
+
+    userobj = model.User.get(user)
+
+    member = (model.Session.query(model.Member)
+              .filter(model.Member.table_name == "user")
+              .filter(model.Member.table_id == userobj.id)
+              .filter(model.Member.group_id == group.id).first())
+
+    # If there is a member for this organization and it is NOT deleted. Reuse
+    # existing if deleted
+    if member:
+        if member.state == 'pending':
+            message = _(
+                "You already have a pending request to the organization")
+        elif member.state == 'active':
+            message = _("You are already part of the organization")
+        # Unknown status. Should never happen..
+        elif member.state != 'deleted':
+            message = _("Duplicate organization request")
+            raise logic.ValidationError({"organization": message}, {_("Organization"): message})
+    else:
+        member = model.Member(table_name="user", table_id=userobj.id, group=group,
+                              group_id=group.id, capacity=role, state='pending')
+
+    # TODO: Is there a way to get language associated to all admins. User table there is nothing as such stored
+    locale = get_safe_locale()
+
+    member.state = 'pending'
+    member.capacity = role
+    member.group = group
+
+    logging.warning("Member's Group ID: " + str(type(member.group_id)))
+    logging.warning(repr(member))
```

Comment:
Are these really valid warning logs?

Reply:
This PR required to duplicate a bit of code form ytp-request. I didn't want to focus on any other places then just where we're injecting custom unaids logic.